### ISSUE TITLE
Status 404 for existing pages

### ISSUE THREAD

**Issue by dragomano** [withastro/starlight#1975]
### What version of `starlight` are you using?

0.24.1

### What version of `astro` are you using?

4.10.1

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### What browser are you using?

Firefox

### Describe the Bug

Startlight 0.23.4:
![sshot-10](https://github.com/withastro/starlight/assets/229402/5bf528cc-1add-41be-acfc-ccd6dc103b69)

Starlight 0.24.0/0.24.1:
![sshot-17](https://github.com/withastro/starlight/assets/229402/4de176e5-dd12-42ac-9233-db8f078babc4)

In both cases, the pages exist and are displayed.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-ztcekb

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for reporting this @dragomano — I think this is caused because we now use Astro’s i18n API under the hood and I guess that is reporting 404s for those routes even though the exist?

I’m curious if anyone has seen this in other uses of Astro’s i18n features? 🤔 

cc @ematipico

**Comment by lorenzolewis** [comment]
I'm curious if this is an issue with `en` being hard-coded in some places in Starlight? Based on the repro, the locale config looks like this:

```ts
{
  defaultLocale: 'root',
    locales: {
      root: {
        label: 'Русский',
        lang: 'ru',
      },
    },
}
```

Some error messages when running `npm run dev`:

```
npm run dev

> @example/starlight-basics@0.0.1 dev
> astro dev

10:08:52 AM [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/utils/i18n.ts:
|- AstroUserError: Failed to get locale informations for the 'en' locale.

10:08:52 AM [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/integrations/shared/localeToLang.ts: failed to import "/node_modules/@astrojs/starlight/utils/i18n.ts"
|- AstroUserError: Failed to get locale informations for the 'en' locale.

10:08:52 AM [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/integrations/expressive-code/translations.ts: failed to import "/node_modules/@astrojs/starlight/integrations/shared/localeToLang.ts"
|- AstroUserError: Failed to get locale informations for the 'en' locale.

10:08:52 AM [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/integrations/expressive-code/index.ts: failed to import "/node_modules/@astrojs/starlight/integrations/expressive-code/translations.ts"
|- AstroUserError: Failed to get locale informations for the 'en' locale.

10:08:52 AM [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/index.ts: failed to import "/node_modules/@astrojs/starlight/integrations/expressive-code/index.ts"
|- AstroUserError: Failed to get locale informations for the 'en' locale.

10:08:52 AM [vite] Error when evaluating SSR module /home/projects/github-ztcekb/astro.config.mjs: failed to import "/node_modules/@astrojs/starlight/index.ts"
|- AstroUserError: Failed to get locale informations for the 'en' locale.

[astro] Unable to load your Astro config

[AstroUserError] Failed to get locale informations for the 'en' locale.
  Hint:
    Make sure to provide a valid BCP-47 tags (e.g. en, ar, or zh-CN).
```

**Comment by HiDeoo** [comment]
> Some error messages when running `npm run dev`:

The error you're seeing is probably related to #1986 and I'm assuming you may have opened the repro using Safari.

I'm currently working on a fix for that one right now.

**Comment by lorenzolewis** [comment]
Ah yup, it was Safari. Once again my testing in the worst browser for JS gets me again 😅

**Comment by ematipico** [comment]
How do I get to see the error from the reproduction? It just fails cc @dragomano

**Comment by lorenzolewis** [comment]
If you’re troubleshooting you should also update to starlight 0.24.1 as it fixed an issue related to this. But it may have not resolved the full issue.

**Comment by dragomano** [comment]
I've updated dependencies. It's still a problem.

**Comment by ematipico** [comment]
Changing `root.lang` to `en` fixes the issue... so maybe there are places where starlight is pulling `en` instead of the configured root lang

**Comment by delucis** [comment]
Fixed in Starlight v0.24.2 — thanks again for reporting!
